security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/OpenIdConnectProfile.scala

```diff
@@ -73,7 +73,11 @@ object OpenIdConnectProfile extends OAuth2Profile[OpenIdConnectUserInfo] {
     val userRoles =
       profile.roles ++
       getUserRoles(profile.subject.get, configuration)
-    val username = profile.preferredUsername.orElse(profile.nickname).orElse(profile.subject).get
+    val username =
+      getUserUsername(profile.subject.get, configuration)
```

Comment:
Can you add some method in `OpenIdConnectUserInfo` to avoid calling `subject.get` in various places and call it once in this place? Also let's add `.getOrElse(throw` at the end of this username determining chain